src/private/types/cli_shorthand.rs

```diff
@@ -0,0 +1,246 @@
+use nom::{
+    branch::alt,
+    bytes::complete::{escaped, is_not, tag, take_till},
+    character::complete::{alpha1, multispace0, one_of},
+    combinator::{eof, map},
+    multi::fold_many1,
+    sequence::{delimited, preceded, separated_pair, terminated},
+    IResult,
+};
+
+///A parser for AWS CLI shorthand structures, particularly those that are used with AVP.
+///
+/// It uses nom to parse strings that essentially JSON like but shorthand, as used as
+/// shorthand structural values in the AWS CLI. This is particularly targeted at the
+/// AVP SDK `PolicyFilter` structure in order to allow the caller to easily specify such
+/// a structure when constraining the set of policies that are managed by the AVP policy
+/// provider.
+///
+/// See the AVP CLI documentation, and the tests below, for more syntax information.
+///
+/// The parser performs no String allocations.
+///
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Value<'src> {
```

Comment:
the type `Value` is used in a lot contexts to represent a value - 

* `serde_json` (of course), along with many of the serde_based crates listed in the serde [documentation](https://docs.rs/serde/latest/serde/#data-formats)
* [simd_json](https://docs.rs/simd-json/latest/simd_json/all.html?search=Value)
* [evalexpr](https://docs.rs/evalexpr/12.0.1/evalexpr/enum.Value.html)

(And yes I was anal enough to spend a few moments ascertaining those and other examples :))

Since, then, the intermediate representation is semantically identical to that of at least the serde examples, I would prefer to leave it as `Value`.

The shorthand syntax is the same as that used in the AVP `lis-policies` `--filter` option of the AWS CLI.